src/Contrib/Jaeger/HttpCollectorExporter.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use InvalidArgumentException;
+use OpenTelemetry\SDK\Trace\Behavior\SpanExporterTrait;
+use OpenTelemetry\SDK\Trace\Behavior\UsesSpanConverterTrait;
+use OpenTelemetry\SDK\Trace\SpanExporterInterface;
+
+class HttpCollectorExporter implements SpanExporterInterface
+{
+    use UsesSpanConverterTrait;
+    use SpanExporterTrait;
+
+    private SpanConverter $spanConverter;
+
+    private ThriftHttpSender $sender;
+
+    public function __construct(
+        $name,
+        string $endpointUrl
+    ) {
+        $parsedDsn = parse_url($endpointUrl);
+
+        if (!is_array($parsedDsn)) {
+            throw new InvalidArgumentException('Unable to parse provided DSN');
+        }
+
+        if (!isset($parsedDsn['host']) || !isset($parsedDsn['port'])) {
+            throw new InvalidArgumentException('Endpoint should have host, port');
+        }
+
+        $this->sender = new ThriftHttpSender(
+            $name,
+            $parsedDsn['host'],
+            $parsedDsn['port'],
+            isset($parsedDsn['path']) ? $parsedDsn['path'] : '', //Matching THttpClient's default
```

Comment:
You can use the short notation (also see below):
```php
$parsedDsn['path'] ?: ''
```

Reply:
Ah actually it seems like Phan is concerned about the key not existing (which may've been why I used the isset and the ternary before now that I try to remember back). 

Is there a way to check for existence but used the condensed form still?

Here's the Phan error output

src/Contrib/Jaeger/HttpCollectorExporter.php:39 PhanTypePossiblyInvalidDimOffset Possibly invalid offset 'path' of $parsedDsn of array type array{port:int,host:string,scheme?:string,user?:string,pass?:string,path?:string,query?:string,fragment?:string}
src/Contrib/Jaeger/HttpCollectorExporter.php:40 PhanTypePossiblyInvalidDimOffset Possibly invalid offset 'scheme' of $parsedDsn of array type array{port:int,host:string,scheme?:string,user?:string,pass?:string,path?:string,query?:string,fragment?:string}

Reply:
@Grunet 
Yes, there is (and that' s what I actually meant to write above):
```php
$parsedDsn['path'] ?? ''
```
The ?? short hand acts like using an `isset` on the left hand operand.

Reply:
Ahh neat! Updated. And remembering this one for sure now :)